bdns/servers.go

```diff
@@ -0,0 +1,137 @@
+package bdns
+
+import (
+	"fmt"
+	"math/rand"
+	"net"
+	"sync"
+	"time"
+)
+
+// serverProvider represents a type which can provide a list of addresses for
+// the bdns to use as DNS resolvers. Different implementations may provide
+// different strategies for providing addresses, and may provide different kinds
+// of addresses (e.g. host:port combos vs IP addresses).
```

Comment:
This seems surprising! I would think it would make more sense to always provide host:port combos.

Reply:
Yeah, I did it this way so that the existing codepath (with staticProvider) could remain exactly the same, i.e. still do the DNS lookups at query time. I wanted to do that just for the transition period -- once the configs are updated to only have a single host each (or maybe once we change the config format to only allow a single host? I haven't touched that *yet* in this change, but maybe I should) then the `staticProvider` would cease to exist and be replaced with a test-only `mockProvider`. Does that make sense?

Reply:
That makes sense!